### PR TITLE
Comments vehicle_move.cpp

### PR DESCRIPTION
**Pull request by mora145**
#### Summary
Infrastructure "Comments are added and improved."

#### Describe the solution
Improve the previous comments and add some missing ones to explain what some functions do. Some arguments are also added.

#### Describe alternatives you've considered
Not applicable

#### Testing
Not applicable

#### Additional context
Not applicable

### CODE REVIEW

File: src/vehicle_move.cpp [hunk 1]
```diff
@@ -569,6 +616,15 @@ void vehicle::thrust( int thd, int z )
...
 }
 
+/**
+ * If the amount is positive, increase the cruise velocity by the amount, but not above the maximum
+ * velocity. If the amount is negative, decrease the cruise velocity by the amount, but not below the
+ * maximum reverse velocity
+ * 
+ * @param amount The amount to change the cruise velocity by.
+ * 
+ * @return the cruise velocity.
```
Comment (BevapDin) [inline src/vehicle_move.cpp:626]: The function does not return anything. It's declared `void`. Same applies to various functions below.

Comment (mora145) [reply src/vehicle_move.cpp:626]: Ok. Thx!

File: src/vehicle_move.cpp [hunk 2]
```diff
@@ -622,6 +680,11 @@ void vehicle::turn( units::angle deg )
     turn_dir = round_to_multiple_of( turn_dir, 15_degrees );
 }
 
+/**
+ * "Stop the vehicle, optionally updating the map cache."
+ * @param update_cache Whether to update the map cache.
+ * @return A reference to the map object.
```
Comment (BevapDin) [inline src/vehicle_move.cpp:686]: The function does not return anything.

File: src/vehicle_move.cpp [hunk 3]
```diff
@@ -145,7 +150,7 @@ void vehicle::smart_controller_handle_turn( bool thrusting,
             c_engines.push_back( i );
         }
     }
-
+    // Checking if the player is flying or is a rotorcraft.
```
Comment (BevapDin) [inline src/vehicle_move.cpp:153]: This refers to `vehicle::is_flying`, the player is completely irrelevant here.

Comment (Fris0uman) [inline src/vehicle_move.cpp:153]:
```suggestion
    // Checking if the vehicle is currently flying or is a rotorcraft.
```

**Comment by Fris0uman** [comment]
closing as stale, feel free to reopen to finish it

**Pull request closed without merging**